src/client/scripts/esm/chess/logic/movesets.ts

```diff
@@ -9,28 +9,48 @@
 
 // Type definitions...
 
+import type { gamefile } from './gamefile.js';
+
 /**
  * A Movesets object containing the movesets for every piece type in a game
- * @typedef {Object<string, PieceMoveset>} Movesets
  */
+type Movesets = {
+	[key: string]: PieceMoveset
+}
+
+/**
+ * The return value of the ignore function
+ */
+interface IgnoreResult {
```

Comment:
So since the blocking function is now separate from the ignore function, I don't think we need this type anymore, as the ignore function and and blocking functions just return a Boolean.